deepnote_toolkit/streamlit_data_apps.py

```
@@ -0,0 +1,385 @@
+        ) from error
+    except urllib.error.URLError as error:
+        raise StreamlitFederatedAuthError(
+            f"Federated auth request failed: {error}"
+        ) from error
+
+    try:
+        payload = json.loads(body)
+    except json.JSONDecodeError as error:
+        raise StreamlitFederatedAuthError(
+            f"Federated auth response was not valid JSON: {body!r}"
+        ) from error
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)